drivers/dac/dac_esp32.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_esp32_dac
+
+#include <soc.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/dac.h>
+#include <zephyr/drivers/clock_control.h>
+#include <hal/rtc_io_types.h>
+#include <hal/rtc_io_hal.h>
+#include <hal/rtc_io_ll.h>
+#include <hal/dac_hal.h>
+#include "driver/dac_common.h"
+
+#include <esp_attr.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(esp32_dac, CONFIG_DAC_LOG_LEVEL);
+
+/* Driver config */
+struct dac_esp32_config {
+	int irq_source;
+	const struct device *clock_dev;
+	clock_control_subsys_t clock_subsys;
+	uint8_t num_channels;
+	uint8_t generator_channel;
+};
+
+/* Driver data */
+struct dac_esp32_data {
+	uint8_t resolution;
+	dac_cw_config_t *generator_cfg;
+};
+
+static int dac_esp32_write_value(const struct device *dev,
+				uint8_t channel, uint32_t value)
+{
+	struct dac_esp32_data *data = dev->data;
+	const struct dac_esp32_config *cfg = dev->config;
+
+	if (data->generator_cfg && channel == conf->generator_channel) {
+		if (value) {
+			dac_cw_generator_config(data->generator_cfg);
+			dac_cw_generator_enable();
+		} else {
+			dac_cw_generator_disable();
+		}
+		return 0;
+	}
+
+	dac_output_voltage(channel, value);
+
+	return 0;
+}
+
+static int dac_esp32_channel_setup(const struct device *dev,
+				   const struct dac_channel_cfg *channel_cfg)
+{
+	struct dac_esp32_data *data = dev->data;
+	const struct dac_esp32_config *cfg = dev->config;
+
+	if (channel_cfg->channel_id > cfg->num_channels) {
+		LOG_ERR("Channel %d is not valid", channel_cfg->channel_id);
+		return -EINVAL;
+	}
+
+	if (channel_cfg->resolution != 8) {
```

Comment:
If this only support 8 bits resolution what is the point of having a field with resolution inside of the config structure? It would be better to remove this field and treat the DAC as 8bits always and just introduce the resolution field when other resolutions get supported.

Reply:
You are right, and there are no other ESP32 SOCs which would support other resolutions, so I removed it, as well as num_channels property which follows same pattern.